### PR TITLE
Add more modes

### PR DESCRIPTION
**Pull request by KornM**

### CODE REVIEW

File: check_hp_msa.py [hunk 1]
```diff
@@ -74,8 +94,8 @@ def __init__(self, api, username, password, logger=None, insecure=False):
         self.insecure = insecure
 
         # TODO: allow debug output
-        #self.debug_outdir = 'tmp/'
-        self.debug_outdir = None
+        self.debug_outdir = 'tmp/'
```
Comment (lazyfrosch) [inline check_hp_msa.py:97]: Should not be enabled by default - and become an option later

File: check_hp_msa.py [hunk 2]
```diff
@@ -403,7 +389,41 @@ def main():
     mode = None
 
     if args.mode == 'disks':
-        mode = client.get_component(Disks, "disks", "drives")
+        output=['location','vendor','model','size','serial-number','status']
+        mode = client.kget_component( "disks", "drives",output)
```
Comment (lazyfrosch) [inline check_hp_msa.py]: What is `kget_component` ?

File: check_hp_msa.py [hunk 3]
```diff
@@ -305,43 +311,42 @@ def build_output(self):
...
+            if self.perfDataList is not None:   
+                for perfDataLine in self.perfDataList:
+                    if isinstance(perfDataLine, str):
+                        self.perfdata.append(f"{self.objectName}_{perfDataLine}={p[perfDataLine]}")
+                    elif isinstance(perfDataLine, list):
+                        usage=perfDataLine[0]
+                        total=perfDataLine[1]
+                        name=perfDataLine[2]
+                        pct=getPct(p[usage],p[total])
+                        self.perfdata.append(f"{self.objectName}_pct_{name}={pct}")
```
Comment (lazyfrosch) [inline check_hp_msa.py:332]: Perfdata label should be cleaned up and special chars replaced by `_`, so we have sth like `number_of_disks`

File: check_hp_msa.py [hunk 4]
```diff
@@ -305,43 +311,42 @@ def build_output(self):
...
+            outputLine=f"[{health}]" 
+            for output in self.outputList:
+                outputLine+=f" {p[output]}"
+            self.output.append(outputLine)
+           
+            # Handling PerfData
+            if self.perfDataList is not None:   
+                for perfDataLine in self.perfDataList:
+                    if isinstance(perfDataLine, str):
+                        self.perfdata.append(f"{self.objectName}_{perfDataLine}={p[perfDataLine]}")
```
Comment (lazyfrosch) [inline check_hp_msa.py:326]: Perfdata label should be cleaned up and special chars replaced by `_`, so we have sth like `number_of_disks`

**Review by lazyfrosch — CHANGES_REQUESTED** [review]
Would be great when all API types follow the same health logic - wasn't sure if this is the case.

Still this would block the way to check certain additional props apart from

Code styling in general can be improved :wink:
- whitespacing
- no `camelCasing` in Python, use [snake case as in PEP8](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

File: check_hp_msa.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (lazyfrosch) [inline check_hp_msa.py:1]: Don't do this, `venv` takes care of setting the proper `python` target. Should only be changed on installation.

File: check_hp_msa.py [hunk 6]
```diff
@@ -101,6 +101,7 @@ def __init__(self, api, username, password, logger=None, insecure=False):
             self.session.verify = not insecure
 
             import urllib3
+            requests.packages.urllib3.disable_warnings()
```
Comment (lazyfrosch) [inline check_hp_msa.py:104]: This can fail in certain environments

**Review by lazyfrosch — CHANGES_REQUESTED** [review]
Few more notes

**Comment by lazyfrosch** [comment]
Do you want me to work on the PR?

It would be nice to add fixtures, so we can add testing as well, similar to: https://github.com/NETWAYS/check_hp_msa/blob/main/test/fixtures/show-disks.xml

Please anonymize!

**Comment by KornM** [comment]
Thanks for the annotations, will work on it as soon as I find some spare
time

Am Mo., 10. Mai 2021 um 17:29 Uhr schrieb Markus Frosch <
***@***.***>:

**Comment by martialblog** [comment]
Marked as stale. Can be reopened with the requested changes anytime.

**Pull request closed without merging**